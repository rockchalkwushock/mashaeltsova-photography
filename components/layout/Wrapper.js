const Wrapper = ({ children }) => (
  <div className="wrapper">
    {children}
    <style jsx>{`
      .wrapper {
        display: -webkit-box;
        display: -ms-flexbox;
        display: flex;
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
        -ms-flex-direction: column;
        flex-direction: column;
        min-height: 100vh;
        text-align: center;
      }
    `}</style>
  </div>
)

export default Wrapper
