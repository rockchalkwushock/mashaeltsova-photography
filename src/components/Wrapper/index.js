import { FlexContainer } from '../commons'

const Wrapper = FlexContainer.extend`
  background-color: ${({ theme }) => theme.bg};
  min-height: 100vh;
  max-width: 100vw;
  padding: 0 1em;
  @media (min-width: ${({ theme }) => theme.sizes.md}) {
    padding: 0 4em;
  }
`

Wrapper.displayName = 'Wrapper'

export default Wrapper
