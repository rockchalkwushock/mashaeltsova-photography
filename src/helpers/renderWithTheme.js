import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
