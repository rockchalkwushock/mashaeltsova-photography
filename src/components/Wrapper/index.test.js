import React from 'react'

import Wrapper from './'
import { renderWithTheme } from '../../helpers'

describe('Styled Component: <Wrapper />', () => {
  test('should render markup', () => {
    const wrapper = renderWithTheme(<Wrapper />)
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    const wrapper = renderWithTheme(<Wrapper />)
    expect(wrapper).toHaveStyleRule('background-color', '#ffffff')
    expect(wrapper).toHaveStyleRule('min-height', '100vh')
    expect(wrapper).toHaveStyleRule('max-width', '100vw')
    expect(wrapper).toHaveStyleRule('padding', '0 1em')
    expect(wrapper).toHaveStyleRule('padding', '0 4em', {
      media: '(min-width:736px)'
    })
  })
})
