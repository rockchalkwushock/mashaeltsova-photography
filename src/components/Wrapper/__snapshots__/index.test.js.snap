// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <Wrapper /> should render markup 1`] = `
.c0 {
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: inherit;
  -webkit-justify-content: inherit;
  -ms-flex-pack: inherit;
  justify-content: inherit;
  text-align: inherit;
  background-color: #ffffff;
  min-height: 100vh;
  max-width: 100vw;
  padding: 0 1em;
}

@media (min-width:736px) {
  .c0 {
    padding: 0 4em;
  }
}

<ThemeProvider
  theme={
    Object {
      "accent": "#df744a",
      "bg": "#ffffff",
      "fg": "#bfd8d2",
      "highlight": "#dcb239",
      "lowlight": "#36454f",
      "sizes": Object {
        "lg": "900px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <Wrapper
    align="inherit"
    direction="row"
    justify="inherit"
    tAlign="inherit"
  >
    <div
      className="c0"
      direction="row"
    />
  </Wrapper>
</ThemeProvider>
`;
